ui/packages/catalog-ui-search/src/main/webapp/component/query-src/query-src.view.js

```diff
@@ -16,12 +16,51 @@
 const Marionette = require('marionette')
 const _ = require('underscore')
 const $ = require('jquery')
-const template = require('./query-src.hbs')
 const CustomElements = require('../../js/CustomElements.js')
 const sources = require('../singletons/sources-instance.js')
+const properties = require('properties')
+import * as React from 'react'
+import { IntlProvider, FormattedMessage } from 'react-intl'
 
 module.exports = Marionette.ItemView.extend({
-  template: template,
+  template(props) {
+    return (
+      <React.Fragment>
+        <div className="choice is-all is-available">
+          <span className="choice-text">
+            <IntlProvider locale="en">
+              <FormattedMessage
+                id="sourcesKeyword"
+                defaultMessage={`All {sources}`}
+                values={{ sources: properties.keywords['Sources'] }}
+              />
+            </IntlProvider>
+          </span>
+          <span className="choice-selected fa fa-check" />
+        </div>
+        {props.sources.map(source => (
+          <div
+            className={
+              'choice is-specific ' + (source.available ? 'is-available' : '')
+            }
+            data-value={source.id}
+          >
+            <span>
+              {!source.available ? (
```

Comment:
✏️ Might be a little more clear to pull some of this conditional rendering up into static methods